### PR TITLE
Additional StrategyManager test coverage

### PR DESCRIPTION
**Pull request by 8sunyuan**
Description:

Additional testing coverage as [mentioned here](https://github.com/Layr-Labs/eigenlayer-contracts/pull/224#issuecomment-1759628467)
Updated the testing excel doc with tests for `addShares()`, `removeShares()`, and `withdrawSharesAsTokens()`
and removed the outdated commented out tests.

### CODE REVIEW

File: src/test/unit/StrategyManagerUnit.t.sol [hunk 1]
```diff
@@ -1075,27 +830,260 @@ contract StrategyManagerUnitTests is Test, Utils {
...
+        IStrategy[] memory strategies = new IStrategy[](3);
+        strategies[0] = dummyStrat;
+        strategies[1] = dummyStrat2;
+        strategies[2] = dummyStrat3;
+        IStrategy removeStrategy = strategies[randStrategy % 3];
+        uint256 removeAmount = amounts[randStrategy % 3];
+        _depositIntoStrategySuccessfully(strategies[0], staker, amounts[0]);
+        _depositIntoStrategySuccessfully(strategies[1], staker, amounts[1]);
+        _depositIntoStrategySuccessfully(strategies[2], staker, amounts[2]);
+
```
Comment (ChaoticWalrus) [inline src/test/unit/StrategyManagerUnit.t.sol:949]: could put some of this in a loop, but it's fine as-is honestly

**Review by ChaoticWalrus — APPROVED** [review]
LGTM

**Review by stevennevins — APPROVED** [review]

**Comment by stevennevins** [comment]
lgtm

File: src/test/unit/StrategyManagerUnit.t.sol (hunk 1 above)
Comment (8sunyuan) [reply src/test/unit/StrategyManagerUnit.t.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/249/commits/c2eb3ff71905ebd4c3293471201e204e6f54f7f1 Looops

**Pull request merged**